config/constants/settings/definition.rb

```diff
@@ -163,56 +1154,34 @@ def add(name,
                          writable:,
                          allowed:,
                          env_alias:)
-
         override_value(definition)
-
-        all << definition
+        @all[name] = definition
       end
 
       def define(&)
         instance_exec(&)
       end
 
       def [](name)
-        by_name[name.to_s]
+        name = name.to_sym
+        if @all.present? && @all.key?(name)
+          @all[name]
+        else
+          h = DEFINITIONS[name]
+          add(name, **h) if h.present?
+        end
       end
 
       def exists?(name)
-        by_name.keys.include?(name.to_s)
-      end
-
-      def all_of_prefix(prefix)
-        all.select { |definition| definition.name.start_with?(prefix) }
+        @all.keys.include?(name.to_sym)
```

Comment:
Also I suspect this method will not work until at least one definition is loaded, as it would be `nil`. The problem would go away if using `all` with

```
def all
  @all ||= {}
end
```

Reply:
Great idea! Applied.